homeworks/hw4/task2.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+)
+
+func main() {
+	nums := []int{2, 3, 4, 5, 6, 7, 8, 9, 10}
+	primeNums := make([]int, 0)
+	compositeNums := make([]int, 0)
+
+	primeChan := goNums(nums, true)
+	compositeChan := goNums(nums, false)
+
+	wg := &sync.WaitGroup{}
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		for {
+			select {
+			case num, ok := <-primeChan:
+				if !ok {
+					return
+				}
+				primeNums = append(primeNums, num)
+			}
+		}
```

Comment:
можно было бы попроще через цикл для чтения из канала прописать, когда канал закроется, цикл завершится
```suggestion
		for prime := range primeChan {
			primeNums = append(primeNums, prime)
		}
```